src/libpmempool/pool.c

```diff
@@ -384,10 +384,15 @@ pool_params_parse(const PMEMpoolcheck *ppc, struct pool_params *params,
 		memcpy(&pbp, addr, sizeof(pbp));
 		params->blk.bsize = le32toh(pbp.bsize);
 	} else if (params->type == POOL_TYPE_OBJ) {
-		struct pmemobjpool pop;
```

Comment:
This seems unnecessary.
If I understand the code correctly, the same can be achieved simply by doing the following:
```
struct pmemobjpool *pop = addr;
memcpy(params->obj.layout, pop.layout, PMEMOBJ_MAX_LAYOUT);
```